plugin/hooks.go

```diff
@@ -149,4 +154,43 @@ type Hooks interface {
 	// Note that this method will be called for files uploaded by plugins, including the plugin that uploaded the post.
 	// FileInfo.Size will be automatically set properly if you modify the file.
 	FileWillBeUploaded(c *Context, info *model.FileInfo, file io.Reader, output io.Writer) (*model.FileInfo, string)
+
+	// PushNotificationWillBeSent is invoked before the push notification has been sent to the push proxy server.
+	// You should return the notification that you want to be sent to the user (normally the same notification).
+	//
+	// You can reject the notification returning a nil notification.
+	//
+	// You can modify the notification, returning a changed version of the notification.
+	//
+	// Minimum server version: 5.10
+	PushNotificationWillBeSent(c *Context, notification *model.PushNotification) *model.PushNotification
+
+	// PushNotificationHasBeenSent is invoked after the push notification has been sent to the push proxy server.
+	//
+	// Minimum server version: 5.10
+	PushNotificationHasBeenSent(c *Context, notification *model.PushNotification)
+
+	// PushNotificationHasFailed is invoked after the push proxy server has replied with an error.
+	//
+	// Minimum server version: 5.10
+	PushNotificationHasFailed(c *Context, notification *model.PushNotification, err error)
+
+	// PushNotificationEnqueued is invoked just before the notification has been enqueued to be sent to the push proxy server.
+	//
+	// This is an internal hook and it can change anytime in the future.
+	//
+	// Minimum server version: 5.10
+	PushNotificationEnqueued(c *Context, notificationId, notificationType, userId, channelId, postId string)
+
+	// PushNotificationAck is invoked when the server receive the confirmation
+	// of the reception of the push notification from the mobile device.
+	//
+	// You can return a notification that will be return to the mobile device
```

Comment:
is your thinking here to use this endpoint to get the data of the notification in case we only send the notification id to tighten security? cause if this is the case at least for iOS a flag needs to be set in the original notification payload

Reply:
Yes, that was my thinking, but I'm open to other options. This is an exploratory PR that needs the mobile implementation details to makes sense entirely, so... whatever you think is better for the mobile apps, is the correct option for me :)

Reply:
Then it does make sense, probably it will need a few more tweaks but it is the right approach